src/realm/column.hpp

```diff
@@ -1788,11 +1826,29 @@ const ColumnDataType ColumnRandIterator<ColumnDataType>::operator*() const
 }
 
 template <class ColumnDataType>
+const ColumnDataType ColumnRandIterator<ColumnDataType>::operator->() const
+{
+    return col->get(col_ndx);
+}
+
+template <class ColumnDataType>
+const ColumnDataType ColumnRandIterator<ColumnDataType>::operator[](const ptrdiff_t& offset) const
```

Comment:
There's no reason to pass a const reference all these places. It only adds visual clutter